tests/integration_tests/test_async_download.py

```diff
@@ -0,0 +1,93 @@
+import io
+import uuid
+from datetime import datetime
+from unittest.mock import Mock
+
+import pytest
+from notifications_python_client.notifications import NotificationsAPIClient
+
+from config import Config
+from core.aws import _S3_CLIENT, upload_file
+from core.const import DATETIME_ISO_8601, EXCEL_MIMETYPE
+
+
+@pytest.fixture()
+def uploaded_exel_mock_file(seeded_test_client):
+    """Uploads a mock Excel file and deletes it on tear down."""
+    fake_file = io.BytesIO(b"0x01010101")
+    key = datetime.now().strftime(DATETIME_ISO_8601)
+    metadata = {"filename": "fake_file.xlsx"}
+    _S3_CLIENT.put_object(
+        fake_file,
+        Config.AWS_S3_BUCKET_FIND_DATA_FILES,
+        key,
+        ExtraArgs={"Metadata": metadata, "ContentType": EXCEL_MIMETYPE},
+    )
+    yield
+    _S3_CLIENT.delete_object(Bucket=Config.AWS_S3_BUCKET_FIND_DATA_FILES, Key=key)
+
+
+@pytest.fixture()
+def uploaded_json_mock_file(seeded_test_client, test_buckets):
+    """Uploads a mock JSON file and deletes it on tear down."""
+    fake_file = io.BytesIO(b"0x01010101")
+    key = f"fund-monitoring-data-{datetime.now().strftime(DATETIME_ISO_8601)}"
+    metadata = {"filename": "fake_file.json"}
+    _S3_CLIENT.put_object(
+        fake_file,
+        Config.AWS_S3_BUCKET_FIND_DATA_FILES,
+        key,
+        ExtraArgs={"Metadata": metadata, "ContentType": "application/json"},
+    )
+    yield
+    _S3_CLIENT.delete_object(Bucket=Config.AWS_S3_BUCKET_FIND_DATA_FILES, Key=key)
+
+
+def test_upload_file(test_buckets):
+    """
+    GIVEN a file upload to S3 is attempted
+    WHEN it is successful
+    THEN the function should return True
+    """
+    uploaded_file = io.BytesIO(b"report")
+    key = f"fund-monitoring-data-{datetime.now().strftime(DATETIME_ISO_8601)}"
+    upload_success = upload_file(
+        uploaded_file, Config.AWS_S3_BUCKET_FIND_DATA_FILES, datetime.now().strftime(DATETIME_ISO_8601)
+    )
+    assert upload_success
+    _S3_CLIENT.delete_object(Bucket=Config.AWS_S3_BUCKET_FIND_DATA_FILES, Key=key)
+
+
+@pytest.fixture()
+def app_context(test_session):
+    with test_session.application.app_context:
+        yield
+
+
+@pytest.fixture()
+def mocked_notify_success():
+    notifications_client = Mock(spec=NotificationsAPIClient)
+    notifications_client.send_email_notification.return_value = (
+        "success",
+        200,
+    )
+
+
+def test_send_email_url_success(mocked_notify_success):
+    current_datetime = datetime.now().strftime(DATETIME_ISO_8601)
+    notifications_client = Mock(spec=NotificationsAPIClient)
+    notifications_client.send_email_notification(
+        email_address="testing@levellingup.gov.uk",
+        template_id=str(uuid.uuid4()),
+        personalisation={
+            "download_url": f"fund-monitoring-data-{current_datetime}",
+            "find_service_url": "fake_url",
+            "file_format": "xlsx | JSON",
+            "file_size": "11kb",
+        },
+    )
```

Comment:
This seems to just test that the Notify client works correctly all by itself - I don't think there's a huge amount of value here, as the Notify library already has a suite of tests assuring its functionality.

Reply:
Test removed.